pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1922,12 +1922,12 @@ func buildListener(opts buildListenerOpts) *listener.Listener {
 				}
 			}
 			if !fullWildcardFound {
-				sort.Strings(chain.sniHosts)
+				sort.Stable(sort.StringSlice(chain.sniHosts))
```

Comment:
We haven't discussed this yet, but this might be affected as well.

Reply:
I agree stable sort will fix this case but it still seems like we have a fundamental race here which may lead to future bugs down the line (or already does!). Would it make more sense to just copy these slices?

Reply:
The question remaining is: where to copy the slice? When copying right here we might be copying the "wrong" state already. There are various places where SNI hosts and destination CIDRs are written. The ones I peeked at didn't seem particularly suspicious, but I'm not familiar with Istios code.

Reply:
can we do the sorting in `initVirtualServices` so this is once for all and no need copy

Reply:
Is the sni hosts list always copied directly from virtual service without adding or removing any items?

Reply:
Also from gateway, but we have sorted it there.  `GetSNIHostsForServer`